pkg/controller/jobframework/reconciler.go

```diff
@@ -820,7 +820,13 @@ func (r *JobReconciler) stopJob(ctx context.Context, job GenericJob, wl *kueue.W
 	}
 
 	if jws, implements := job.(ComposableJob); implements {
-		stoppedNow, err := jws.Stop(ctx, r.client, info, stopReason, eventMsg)
+		reason := stopReason
+		if stopReason == StopReasonWorkloadEvicted {
+			if evCond := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadEvicted); evCond != nil && evCond.Status == metav1.ConditionTrue {
+				reason = StopReason(string(StopReasonWorkloadEvicted) + "Reason" + evCond.Reason)
```

Comment:
```suggestion
				reason = StopReason(string(StopReasonWorkloadEvicted) + "DueTo" + evCond.Reason)
```
Maybe it reads better?

Reply:
Ok. It definitely reads better.